internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package netapp
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumequotarules"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/netapp/2022-05-01/volumes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	netAppModels "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/models"
+	netAppValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NetAppVolumeQuotaRuleResource struct{}
+
+var _ sdk.Resource = NetAppVolumeQuotaRuleResource{}
+
+func (r NetAppVolumeQuotaRuleResource) ModelObject() interface{} {
+	return &netAppModels.NetAppVolumeQuotaRuleModel{}
+}
+
+func (r NetAppVolumeQuotaRuleResource) ResourceType() string {
+	return "azurerm_netapp_volume_quota_rule"
+}
+
+func (r NetAppVolumeQuotaRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumequotarules.ValidateVolumeQuotaRuleID
+}
+
+func (r NetAppVolumeQuotaRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeQuotaRuleName,
+		},
+
+		"location": commonschema.Location(),
+
+		"volume_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeID,
+		},
+
+		"quota_type": {
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(volumequotarules.PossibleValuesForType(), false),
+		},
+
+		"quota_target": {
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"quota_size_in_kib": {
```

Comment:
Let's move this block up one so all the Required attributes are above the Optional ones